can/interfaces/ixxat/canlib.py

```diff
@@ -40,6 +45,25 @@
     log.warning("IXXAT VCI library does not work on %s platform", sys.platform)
     _canlib = None
 
+def __vciFormatErrorExtended(library_instance, function, HRESULT, arguments):
+    """ Format a VCI error and attach failed function, decoded HRESULT and arguments
+        @TODO: make sure we don't generate another exception
+    """
+    buf = ctypes.create_string_buffer(VCI_MAX_ERRSTRLEN)

```

Comment:
To avoid repeating the same code just call `__vciFormatError` here and add the arguments to the returned string.